projects/dsp-ui/src/lib/viewer/operations/add-value/add-value.component.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    Component,
+    EventEmitter,
+    Inject,
+    Input,
+    OnInit,
+    Output,
+    ViewChild
+} from '@angular/core';
+import {
+    Constants,
+    CreateValue,
+    KnoraApiConnection,
+    ReadResource,
+    ResourcePropertyDefinition,
+    UpdateResource,
+    WriteValueResponse,
+} from '@dasch-swiss/dsp-js';
+import { mergeMap } from 'rxjs/operators';
+import { DspApiConnectionToken } from '../../../core/core.module';
+import { EmitEvent, EventBusService, Events } from '../../services/event-bus.service';
+import { BaseValueComponent } from '../../values/base-value.component';
+
+@Component({
+  selector: 'dsp-add-value',
+  templateUrl: './add-value.component.html',
+  styleUrls: ['./add-value.component.scss']
+})
+export class AddValueComponent implements OnInit {
+
+    @ViewChild('createVal') createValueComponent: BaseValueComponent;
+
+    @Input() resourcePropertyDefinition: ResourcePropertyDefinition;
+
+    @Input() parentResource: ReadResource;
+
+    @Input() configuration?: object;
+
+    @Output() operationCancelled = new EventEmitter<any>();
+
+    constants = Constants;
+
+    mode: 'read' | 'update' | 'create' | 'search';
+
+    canModify: boolean;
+
+    createModeActive = false;
+
+    submittingValue = false;
+
+    // 0 will display a loading animation
+    progressIndicatorStatus = 0;
+
+    progressIndicatorColor = 'blue';
+
+    constructor(@Inject(DspApiConnectionToken)
+                private knoraApiConnection: KnoraApiConnection,
+                private _eventBusService: EventBusService) { }
+
+    ngOnInit() {
+
+        this.mode = 'create';
+
+        this.createModeActive = true;
+
+        // TODO: find a way to figure out what type of text value it is
+        if (this.resourcePropertyDefinition.objectType === 'http://api.knora.org/ontology/knora-api/v2#TextValue') {
+            this.resourcePropertyDefinition.objectType = 'ReadTextValueAsString';
```

Comment:
Yes, the problem is that they all share the object type `http://api.knora.org/ontology/knora-api/v2#TextValue`. To be honest, I haven't really thought about this too much since we don't have edit components made for the other two text value types.

Reply:
Ok, got it. We kind of has the same problem for the value edit case.